custom/icds/management/commands/populate_mother_name.py

```diff
@@ -144,10 +144,10 @@ def _update_cases(self, filename):
                     updates[row['Case ID']] = row['Mother Name']
                     counter += 1
                     if counter > 0 and counter % 100 == 0:
-                        case_ids_list = self._reassured_case_ids_to_update(updates.keys())
-                        for case_id in updates:
-                            if case_id not in case_ids_list:
-                                updates.pop(case_id)
+                        case_ids_list = self._reassured_case_ids_to_update(list(updates.keys()))
+                        skip_ids = set(updates.keys()) - set(case_ids_list)
```

Comment:
nit: `dict.keys` returns a set-like collection. It does not need to be converted to a set
```suggestion
                        skip_ids = updates.keys() - set(case_ids_list)
```

Example:
```py
>>> x = {1: 2, 3: 4}
>>> x.keys() - {3}
{1}
```

nit 2: `_reassured_case_ids_to_update` builds a list, which is then converted to a set on this line. It could instead build a `set` direclty, thereby avoiding the list -> set conversion.

Reply:
thanks @millerdev 
i was trying to retain order of case ids, which came in handy when updates failed mid way and i could locate the case id in the original list of ids to be updated. Updating since things should be working smoothly now since the command has been fixed.